src/WordPress/Blueprints/Runner/Step/UnzipStepRunner.php

```diff
@@ -2,21 +2,32 @@
 
 namespace WordPress\Blueprints\Runner\Step;
 
+use WordPress\Blueprints\BlueprintException;
 use WordPress\Blueprints\Model\DataClass\UnzipStep;
 use WordPress\Blueprints\Progress\Tracker;
 use function WordPress\Zip\zip_extract_to;
 
 class UnzipStepRunner extends BaseStepRunner {
 
+	/**
+	 * Runs the Unzip Step
+	 *
+	 * @param UnzipStep $input Step.
+	 * @param Tracker   $tracker Tracker.
+	 * @return void
+	 * @throws BlueprintException
+	 */
 	public function run(
 		UnzipStep $input,
-		Tracker $progress = null
+		Tracker $tracker
 	) {
-		$progress?->set( 10, 'Unzipping...' );
+		$tracker->set( 10, 'Unzipping...' );
 
-		// @TODO: Expose a generic helper method for this, e.g. $this->getExecutionContext()->resolvePath($input->extractToPath);
-		$toPath = $this->getRuntime()->getDocumentRoot() . '/' . $input->extractToPath;
-		zip_extract_to( $this->getResource( $input->zipFile ), $toPath );
+		$resolved_to_path = $this->getRuntime()->resolvePath( $input->extract_to_path );
+		try {
+			zip_extract_to( $this->getResource( $input->zip_file ), $resolved_to_path );
+		} catch ( \Throwable $exception ) {
```

Comment:
Not a blocker, but: I thought these try/catches would be handy in step runners, but in practice it's such a repetitive piece of boilerplate. It's too easy to forget about it or get it wrong in a single instance, plus it will be a pain to maintain. Let's just be optimistic about the execution result inside step runners and not rethrow BlueprintExceptions, and then catch and handle all exceptions in BlueprintRunner.

Would you mind starting a new issue to track this?

Reply:
Sure. So this PR will not add the try-catch block for the Unzip Step Runner. I'll start a new issue.